### PR TITLE
initial ctop formula

### PR DESCRIPTION
**Pull request by avelino**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

https://github.com/bcicen/ctop/issues/5

### CODE REVIEW

**Comment by bfontaine** [comment]
For the future pull-requests please [follow our guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) and use the commit format `ctop 0.4.1 (new formula)`. Thanks!

File: Formula/ctop.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Ctop < Formula
+  desc "Top-like interface for container metrics"
+  homepage "https://bcicen.github.io/ctop/"
+  url "https://github.com/bcicen/ctop/releases/download/v0.4.1/ctop-0.4.1-darwin-amd64"
```
Comment (bfontaine) [inline Formula/ctop.rb]: We need to build from source here.

File: Formula/ctop.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  sha256 "884555d6303652ba7892ee90faa6527bacbfb73b84c1006edd8916bafb5de22f"
+
+  bottle :unneeded
+
+  def install
+    mv "ctop-0.4.1-darwin-amd64", "ctop"
+    bin.install "ctop"
```
Comment (bfontaine) [inline Formula/ctop.rb]: Don’t use `mv` + `bin.install`. See my comment on your other PR: https://github.com/Homebrew/homebrew-core/pull/10847#discussion_r105453276

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/ctop.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  depends_on 'go'
+  bottle :unneeded
+
+  def install
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/bcicen").mkpath
+    ln_s buildpath, buildpath/"src/github.com/bcicen/ctop"
+    system "go", "build", "-o", "ctop"
+    bin.install "ctop"
```
Comment (bfontaine) [inline Formula/ctop.rb]: You can merge this line and the one above by using `system "go", "build", "-o", bin/"ctop"`.

**Review by bfontaine — APPROVED** [review]

**Comment by avelino** [comment]
Awaiting project solve that problem to send update for homebrew: https://github.com/bcicen/ctop/issues/33

PR on ctop to fixed problem: https://github.com/bcicen/ctop/pull/34 (awaiting approval)

**Comment by fxcoudert** [comment]
```
  * `bottle modifier` (line 8) should be put before `depends_on` (line 7)
  * C: 7: col 14: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
```

File: Formula/ctop.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+class Ctop < Formula
+  desc "Top-like interface for container metrics"
+  homepage "https://bcicen.github.io/ctop/"
+  url "https://github.com/bcicen/ctop/archive/v0.4.1-deps.tar.gz"
```
Comment (bfontaine) [inline Formula/ctop.rb]: This `-deps` confuses the version parser; please add the version below this line using `version "0.4.1"`.

```
Error: 1 problem in 1 formula
ctop:
  * Stable: version (deps) is set to a string without a digit
```

**Comment by avelino** [comment]
@bcicen pls

**Comment by bcicen** [comment]
@avelino - as @bfontaine's comment suggests, all you need to do is insert `version "0.4.1"` below line 4 in the current formula

**Comment by avelino** [comment]
@bcicen 
Formula:
```
class Ctop < Formula
  desc "Top-like interface for container metrics"
  homepage "https://bcicen.github.io/ctop/"
  version "0.4.1"
  url "https://github.com/bcicen/ctop/archive/v0.4.1-deps.tar.gz"
  sha256 "bc10b774dad0bc7ef0be41bcfb36c774fc28dafa789b2f43f1ecdb5b75390867"

  depends_on "go" => :build
  depends_on "glide" => :build

  def install
    ENV["GOPATH"] = buildpath
    ENV["GLIDE_HOME"] = HOMEBREW_CACHE/"glide_home/#{name}"
    dir = buildpath/"src/github.com/bcicen/ctop"
    dir.install Dir["*"]
    cd dir do
      system "glide", "install"
      system "go", "build", "-o", bin/"ctop"
    end
  end

  test do
    system "#{bin}/ctop", "-v"
  end
end
```

Error:
```
$ brew install --build-from-source Formula/ctop.rb
==> Downloading https://github.com/bcicen/ctop/archive/v0.4.1-deps.tar.gz
Already downloaded: /Users/avelino/Library/Caches/Homebrew/ctop-0.4.1.tar.gz
==> glide install
Last 15 lines from /Users/avelino/Library/Logs/Homebrew/ctop/01.glide:
2017-03-13 11:11:42 -0300

glide
install

[ERROR] Failed to find glide.yaml file in directory tree: Cannot resolve parent of /

Do not report this issue to Homebrew/brew or Homebrew/core!

These open issues may also help:
initial ctop formula https://github.com/Homebrew/homebrew-core/pull/10846
```

**Comment by avelino** [comment]
@bcicen Will generate new release?

**Comment by bcicen** [comment]
@avelino you need to clear your local cache before testing the brew install again to fetch the correct version:
```bash
rm -rvf /Users/avelino/Library/Caches/Homebrew/ctop*
```

**Comment by bcicen** [comment]
@avelino version needs to be on line 5, below url. Test error:
```* `url` (line 5) should be put before `version` (line 4)```

**Comment by fxcoudert** [comment]
Thanks @avelino for your contribution to Homebrew!

**Pull request closed without merging**